src/Sarif/NotYetAutoGenerated/Location.cs

```diff
@@ -45,11 +45,6 @@ public virtual SarifNodeKind SarifNodeKind
         [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
         public virtual BigInteger Id { get; set; }
 
-        public bool ShouldSerializeId()
-        {
-            return Id >= 0;
-        }
```

Comment:
The better place for this is to put in core folder, this is not related to this change, moved btw.

Reply:
Yes! But see my other comment. I wondered why we preserved this should serialize id if we have 'emit default value' == false.

One reason is that someone could set this number to a garbage value like -37, which wouldn't be removed by the serialization attribute.

If that's true, though, we don't really need the default value and emit default value attributes. 

But perhaps we do, because someone could emit serialized json where the -1 is explicit (that's legal!) but we don't want to populate in this case.

OK, so talking it all over, I agree with your implementation. :) We need to preserve this file as it is and retain (and move) ShouldSerializeId().

Thanks for listening/. :)

Reply:
why we preserved this should serialize id if we have 'emit default value' == false.
------ I think we need both because 
ShouldSerializeXXX is from Newtonsoft.Json and used in Newtonsoft.Json.JsonConvert
EmitDefaultValue in the DataMember is used in dotnet e.g. DataContractSerializer
I tried removing either one of them will make the corresponding one populate id:-1 value in the file.